tests/unit/synapseclient/extensions/schema_files/data_models/example.model.csv

```diff
@@ -1,84 +1,84 @@
-Check Match at Least values,,,,,TRUE,DataProperty,,,matchAtLeastOne MockComponent.checkMatchatLeastvalues value,,
-Check Match Exactly values,,,,,TRUE,DataProperty,,,matchExactlyOne MockComponent.checkMatchExactlyvalues value,,
-Check Match None values,,,,,TRUE,DataProperty,,,matchNone MockComponent.checkMatchNonevalues value error,,
-Check Recommended,,,,,FALSE,DataProperty,,,recommended,,
-Check Ages,,,,,TRUE,DataProperty,,,protectAges,,
-Check Unique,,,,,TRUE,DataProperty,,,unique error,,
-Check Range,,,,,TRUE,DataProperty,,,inRange 50 100 error,,
-Check Date,,,,,TRUE,DataProperty,,,date,string,date
-Check NA,,,,,TRUE,DataProperty,,,int::IsNA,,
-MockRDB,,,"Component, MockRDB_id, SourceManifest",,FALSE,DataType,,,,,
-MockRDB_id,,,,,TRUE,DataProperty,,,int,,
-SourceManifest,,,,,TRUE,DataProperty,,,,,
-MockFilename,,,"Component, Filename",,FALSE,DataType,,,,,
-JSONSchemaComponent,Component to hold attributes for testing JSON Schemas,,"Component, No Rules, No Rules Not Required, String, String Not Required, Enum, Enum Not Required, Date, URL, InRange, Regex, List, List Not Required, List Enum, List Enum Not Required, List Boolean, List, Integer, List InRange",,FALSE,DataType,,,,,
-No Rules,,,,,TRUE,DataProperty,,,,,
-No Rules Not Required,,,,,FALSE,DataProperty,,,,,
-String,,,,,TRUE,DataProperty,,,,string,
-String Not Required,,,,,FALSE,DataProperty,,,,string,
-Enum,,"ab, cd, ef, gh",,,TRUE,DataProperty,,,,string,
-Enum Not Required,,"ab, cd, ef, gh",,,FALSE,DataProperty,,,,string,
-Date,,,,,TRUE,DataProperty,,,date,string,date
-URL,,,,,TRUE,DataProperty,,,url,string,uri
-InRange,,,,,TRUE,DataProperty,,,inRange 50 100,number,
-Regex,,,,,TRUE,DataProperty,,,regex search [a-f],string,
-List,,,,,TRUE,DataProperty,,,,string_list,
-List Not Required,,,,,FALSE,DataProperty,,,,string_list,
-List Enum,,"ab, cd, ef, gh",,,TRUE,DataProperty,,,,string_list,
-List Enum Not Required,,"ab, cd, ef, gh",,,FALSE,DataProperty,,,,string_list,
-List Boolean,,,,,TRUE,DataProperty,,,,boolean_list,
-List Integer,,,,,TRUE,DataProperty,,,,integer_list,
-List InRange,,,,,TRUE,DataProperty,,,inRange 50 100,integer_list,
-TypeDefinitionComponent,Component to check type specification,,"Component, String type, String type caps, Int type, Int type caps, Num type, Num type caps, Nan type, Missing type, Boolean type, Boolean type caps",,FALSE,DataType,,,,,
-String type,,,,,TRUE,DataProperty,,,,string,
-String type caps,,,,,TRUE,DataProperty,,,,STRING,
-Int type,,,,,TRUE,DataProperty,,,,integer,
-Int type caps,,,,,TRUE,DataProperty,,,,INTEGER,
-Num type,,,,,TRUE,DataProperty,,,,number,
-Num type caps,,,,,TRUE,DataProperty,,,,NUMBER,
-Nan type,,,,,TRUE,DataProperty,,,,nan,
-Missing type,,,,,TRUE,DataProperty,,,,,
-Boolean type,,,,,TRUE,DataProperty,,,,boolean,
-Boolean type caps,,,,,TRUE,DataProperty,,,,BOOLEAN,
+Attribute,Description,Valid Values,DependsOn,Properties,Required,Parent,DependsOn Component,Source,Validation Rules,columnType,Format,Pattern
```

Comment:
I noticed that you modified the original attributes in MockComponent, such as Check Regex List. However, it might be better to add new components under either JSONSchema or a separate component so we can maintain backward compatibility with the existing validation rules and tests while testing the new behavior. What do you think @andrewelamb

Reply:
@linglp Sorry I missed your comment. I think this is fine. The validation rules are only used (here in the curator extension) for creating JSON Schema.